frontend/src/components/my-courses/my-courses.tsx

```diff
@@ -19,35 +19,50 @@ import {
 import styles from './styles.module.scss';
 
 const MyCourses: FC = () => {
-  const { page, handlePageChange } = usePagination({
-    queryName: 'myCoursesPage',
+  const { page: mentorCoursesPage, handlePageChange: handleMentorCoursesPage } =
+    usePagination({
+      queryName: 'mentorCoursesPage',
+    });
+
+  const {
+    page: studentCoursesPage,
+    handlePageChange: handleStudentCoursesPage,
+  } = usePagination({
+    queryName: 'studentCoursesPage',
   });
 
   const dispatch = useAppDispatch();
   const {
     coursesStudying,
     coursesMentoring,
     totalCoursesMentoring,
+    totalCoursesStudying,
     dataStatus,
   } = useAppSelector((state) => ({
     coursesStudying: state.myCourses.coursesStudying,
     coursesMentoring: state.myCourses.coursesMentoring,
     dataStatus: state.myCourses.dataStatus,
     totalCoursesMentoring: state.myCourses.totalCoursesMentoring,
+    totalCoursesStudying: state.myCourses.totalCoursesStudying,
   }));
 
   useEffect(() => {
-    dispatch(myCoursesActions.getCoursesStudying());
-  }, [dispatch]);
+    dispatch(
+      myCoursesActions.getCoursesMentoring({
+        page: mentorCoursesPage,
+        count: PaginationDefaultValue.DEFAULT_COUNT,
+      }),
+    );
+  }, [mentorCoursesPage]);
 
   useEffect(() => {
     dispatch(
-      myCoursesActions.getCoursesMentoring({
-        page,
+      myCoursesActions.getCoursesStudying({
+        page: studentCoursesPage,
         count: PaginationDefaultValue.DEFAULT_COUNT,
```

Comment:
If I'm not mistaken, during the daily it was told, that we should have 15 courses in `As a student` block, was this requirement changed to 5?

Reply:
15 sounds like something too much - within pagination 15 courses would take about 2 full scrolls of screen, and that's with fact that we have `courses as mentor` section below it - so 5 courses as a limit sounds reasonable
![image](https://user-images.githubusercontent.com/69236539/191111489-bed5940e-e0a9-4535-91ac-2443efcbb701.png)